### PR TITLE
refactor: align segmentation3d with architecture

### PR DESCRIPTION
**Pull request by amadeuszsz**
## Summary

- Add a runtime data preprocessing pipeline to `BaseModel`: on_after_batch_transfer applies `DataPreprocessing`, attached via `BaseModel.set_data_preprocessing(...)` from train/test/deploy entrypoints 
- Decouple framework from output-side shaping: voxel -> point decode, softmax, argmax, and metric reconstruction now live inside the model (`compute_metrics` / `predict_outputs`) rather than in a separate pipeline 
- Refactor PTv3 segmentation: `forward()` returns voxel logits, `compute_metrics` and `predict_outputs` use `inverse`/`origin_segment` for point-level metrics and predictions 
- Refactor FRNet to match the same contract: `forward()` returns (`point_logits`, `*voxel_feats_pyramid`), aligned `compute_metrics` and `predict_outputs` with PTv3 (`pred_labels` + `pred_probs` at point level) 
- Remove the intermediate `Seg3DModel` base class and `structures.py`, FRNet and PTv3 extend `BaseModel` directly
- Make `FrustumRangePreprocessor` a stateless plain class (no `nn.Module`) with a vectorized batched majority-vote 
- Extract `datamodule/collation.py` from `datamodule/base.py`, document the collation strategy contract (concat / stack / index_concat / list) 
- Add `default_runtime` -> `modules/data_preprocessing.yaml` defaults
- Update `train.py`, `test.py`, `deploy.py` to call the single-arg `set_data_preprocessing`
- Expand utilities: `MatchingWeightsLoadReport` + verbose logging in `utils/checkpoints.py` updates to deploy, mlflow helpers, point-cloud structures 
- Add an end-to-end integration test that wires `FrustumRangePreprocessor via `set_data_preprocessing and runs `_shared_step`, replace the deleted `BaseSegmentationModel` tests with PTv3 inline-decode coverage
- Update `docs/framework/design.md`, `docs/contributing/adding-models.md`, `docs/user-guide/configuration.md` to reflect the simplified single-pipeline architecture 

## Checklist

- [ ] You've checked our [contribution overview](https://tier4.github.io/autoware-ml/contributing/contribution-overview/).
- [ ] Your PR follows our [pull request guidelines](https://tier4.github.io/autoware-ml/contributing/pull-request-guidelines/).

### CODE REVIEW

File: autoware_ml/utils/checkpoints.py [hunk 1]
```diff
@@ -16,10 +16,266 @@
...
+        map_location: Device or device string used when loading each checkpoint.
+        device: Optional device to move the model to after all weights are loaded.
+        set_eval: Whether to switch the model into evaluation mode after loading.
+        logger: Logger used for per-checkpoint load reports.
+
+    Returns:
+        Per-checkpoint matching-weight load reports.
+    """
+    weight_paths = (weights,) if isinstance(weights, str | Path) else tuple(weights)
+    reports = tuple(
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/utils/checkpoints.py:205]: https://github.com/tier4/autoware-ml/pull/41/commits/db06446bb10df5e6d6bbc8a40628cbc0463a62d2

File: autoware_ml/datamodule/base.py [hunk 2]
```diff
@@ -138,51 +140,68 @@ class DataModule(L.LightningDataModule, ABC):
...
+    Failure modes:
+        Undeclared keys are dropped, declared-but-missing keys are skipped
+        with a warning, and a wrong strategy for a key's content fails loudly.
     """
 
     def __init__(
         self,
-        stack_keys: Sequence[str] | None = None,
+        collation_map: Mapping[str, CollationStrategy] = {},
         train_transforms: TransformsCompose | None = None,
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/datamodule/base.py:172]: https://github.com/tier4/autoware-ml/pull/41/commits/b4e4aacc1ba2faa8002fe780f0b2643c5c11faf1

File: autoware_ml/transforms/point_cloud/sampling.py [hunk 3]
```diff
@@ -146,23 +146,40 @@ def __init__(
...
+                coordinate in world space.
+            return_displacement: Whether to expose per-point displacement to the
+                voxel center.
+            project_displacement: Whether to project displacement to normals.
         """
-        self.grid_size = grid_size
+        self.grid_size = np.asarray(grid_size, dtype=np.float32)
+        self.hash = self.fnv_hash_vec if hash_type == "fnv" else self.ravel_hash_vec
+        if mode not in {"train", "test"}:
+            raise ValueError(f"Unsupported GridSample mode: {mode}")
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/transforms/point_cloud/sampling.py:184]: https://github.com/tier4/autoware-ml/pull/41/commits/d085d299850ec6d5aa3c47475165b03a69f47aac

File: autoware_ml/transforms/point_cloud/sampling.py [hunk 4]
```diff
@@ -173,29 +190,134 @@ def transform(self, input_dict: dict[str, Any]) -> dict[str, Any]:
...
+        scaled_coord: np.ndarray,
+        min_coord_world: np.ndarray,
+        sort_indices: np.ndarray,
+        inverse: np.ndarray,
+        counts: np.ndarray,
+    ) -> dict[str, Any]:
+        """Apply train-time voxel subsampling."""
+        selection = np.cumsum(np.insert(counts, 0, 0)[:-1]) + (
+            np.random.randint(0, counts.max(), counts.size) % counts
+        )
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/transforms/point_cloud/sampling.py]: https://github.com/tier4/autoware-ml/pull/41/commits/d085d299850ec6d5aa3c47475165b03a69f47aac

File: autoware_ml/models/segmentation3d/backbones/ptv3.py [hunk 5]
```diff
@@ -405,7 +406,12 @@ def forward(self, point: Point) -> Point:
         head_count = self.num_heads
         if not self.enable_flash:
             min_points = int(offset_to_bincount(point.offset).min().item())
-            self.patch_size = max(1, min(self.patch_size_max, min_points))
+            if min_points < self.patch_size_max:
+                raise ValueError(
+                    "PTv3 export mode requires each sample to have at least "
+                    f"{self.patch_size_max} serialized points, but found {min_points}."
+                )
+            self.patch_size = self.patch_size_max
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/models/segmentation3d/backbones/ptv3.py]: https://github.com/tier4/autoware-ml/pull/41/commits/15f5fe9fb95350b7708479ca492a6204a8fd1659
